pkg/github/stargazers.go

```diff
@@ -0,0 +1,147 @@
+package github
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/github-datasource/pkg/models"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/pkg/errors"
+	"github.com/shurcooL/githubv4"
+)
+
+// QueryStargazers is the object representation of the graphql query for retrieving a paginated list of stargazers for a repository
+//
+//	query {
+//		repository(owner: $owner, name: $name) {
+//			stargazers(first: 100, orderBy: {field: STARRED_AT, direction: DESC}, after: $cursor) {
+//				totalCount
+//				pageInfo {
+//					hasNextPage
+//					startCursor
+//					endCursor
+//				}
+//				edges {
+//					starredAt
+//					node {
+//						id
+//						login
+//						name
+//						company
+//						email
+//						url
+//					}
+//				}
+//			}
+//		}
+//	}
+type QueryStargazers struct {
+	Repository struct {
+		Stargazers struct {
+			TotalCount int64
+			PageInfo   models.PageInfo
+			Edges      []Stargazer
+		} `graphql:"stargazers(first: 100, orderBy: {field: STARRED_AT, direction: DESC}, after: $cursor)"`
+	} `graphql:"repository(owner: $owner, name: $name)"`
+}
+
+// Stargazer a GitHub user tied to when they starred the repository
+type Stargazer struct {
+	StarredAt githubv4.DateTime
+	Node      models.User
+}
+
+// StargazerWrapper is a mapping of a GitHub stargazer to what the total count
+// of stars roughly would've been when they starred the repository
+type StargazerWrapper struct {
+	Stargazer
+	StarCount int64
+}
+
+// StargazersWrapper is a list of GitHub wrapped stargazers
+type StargazersWrapper []StargazerWrapper
+
+// Frames converts the list of stargazers to a Grafana DataFrame
+func (a StargazersWrapper) Frames() data.Frames {
+	frame := data.NewFrame(
+		"stargazers",
+		data.NewField("star_count", nil, []int64{}),
+		data.NewField("id", nil, []string{}),
+		data.NewField("login", nil, []string{}),
+		data.NewField("git_name", nil, []string{}),
+		data.NewField("company", nil, []string{}),
+		data.NewField("email", nil, []string{}),
+		data.NewField("url", nil, []string{}),
+		data.NewField("starred_at", nil, []time.Time{}),
+	)
+
+	for _, v := range a {
+		node := v.Stargazer.Node
+
+		frame.InsertRow(
+			0,
+			v.StarCount,
+			node.ID,
+			node.Login,
+			node.Name,
+			node.Company,
+			node.Email,
+			node.URL,
+			v.StarredAt.Time,
+		)
+	}
+
```

Comment:
You can add this to make the visual view preferred.

```suggestion
frame.Meta = &data.FrameMeta{PreferredVisualization: data.VisTypeGraph}
```